tests/test_group.rs

```diff
@@ -15,94 +16,146 @@ fn create_commit_optional_path() {
     let bob_credential_bundle =
         CredentialBundle::new("Bob".into(), CredentialType::Basic, ciphersuite_name).unwrap();
 
+    // Mandatory extensions, will be fixed in #164
+    let lifetime_extension = Box::new(LifetimeExtension::new(60));
+    let mandatory_extensions: Vec<Box<dyn Extension>> = vec![lifetime_extension];
+
     // Generate KeyPackages
     let alice_key_package_bundle = KeyPackageBundle::new(
         ciphersuite_name,
-        &alice_credential_bundle, // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
-        Vec::new(),
+        &alice_credential_bundle,
+        mandatory_extensions.clone(),
     );
 
     let bob_key_package_bundle = KeyPackageBundle::new(
         ciphersuite_name,
-        &bob_credential_bundle, // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
-        Vec::new(),
+        &bob_credential_bundle,
+        mandatory_extensions.clone(),
     );
     let bob_key_package = bob_key_package_bundle.get_key_package();
 
     let alice_update_key_package_bundle = KeyPackageBundle::new(
         ciphersuite_name,
-        &alice_credential_bundle, // TODO: bad API, we shouldn't have to get the private key out here (this function shouldn't exist!)
-        Vec::new(),
+        &alice_credential_bundle,
+        mandatory_extensions,
     );
     let alice_update_key_package = alice_update_key_package_bundle.get_key_package();
+    assert!(alice_update_key_package.verify().is_ok());
 
     // Alice creates a group
     let group_id = [1, 2, 3, 4];
-    let group_alice_1234 = MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
+    let mut group_alice_1234 = MlsGroup::new(&group_id, ciphersuite_name, alice_key_package_bundle);
 
-    // Alice adds Bob
+    // Alice proposes to add Bob with forced self-update
+    // Even though there are only Add Proposals, this should generated a path field on the Commit
     let bob_add_proposal = group_alice_1234.create_add_proposal(
         group_aad,
-        &bob_credential_bundle,
+        &alice_credential_bundle,
         bob_key_package.clone(),
     );
+    let epoch_proposals = vec![bob_add_proposal];
+    let (mls_plaintext_commit, _welcome_bundle_alice_bob_option) = match group_alice_1234
+        .create_commit(
+            group_aad,
+            &alice_credential_bundle,
+            epoch_proposals,
+            true, /* force self-update */
+        ) {
+        Ok(c) => c,
+        Err(e) => panic!("Error creating commit: {:?}", e),
+    };
+    let commit = match &mls_plaintext_commit.content {
+        MLSPlaintextContentType::Commit((commit, _)) => commit,
+        _ => panic!(),
+    };
+    assert!(commit.path.is_some());
```

Comment:
Can we test more here than just the fact that there is a path?

Reply:
The follow-up PR does test more. We need the API change for that.